pkg/ddl/generated_column.go

```diff
@@ -289,6 +289,11 @@ type illegalFunctionChecker struct {
 func (c *illegalFunctionChecker) Enter(inNode ast.Node) (outNode ast.Node, skipChildren bool) {
 	switch node := inNode.(type) {
 	case *ast.FuncCallExpr:
+		// Grouping function is not allowed, issue #49909.
+		if node.FnName.L == ast.Grouping {
```

Comment:
I think it's better to add it to `IllegalFunctions4GeneratedColumns` ?

Reply:
> I think it's better to add it to `IllegalFunctions4GeneratedColumns` ?

`illegalFunctionChecker ` is for that. It parse the `expr` and set flag, then `IllegalFunctions4GeneratedColumns` judge it.